From 2af39b3f35217e6a337fa5d4275aa6ec79cf49d3 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sun, 27 May 2018 23:22:33 +0200
Subject: [PATCH] lineagehw: Enable ReadingEnhancement for HWC1 as well

 * ReadingEnhancement performs properly on HWC1 too
 * Register non-HWC2 mode as HWC1, and allow support

 * Future evolutions may restrict some HWC1 based on
    certain criterias justifying disabling support

Change-Id: I1fa39bedd4aba67895c2a0ea0ded0470e40dda90
---
 src/org/lineageos/hardware/ReadingEnhancement.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/org/lineageos/hardware/ReadingEnhancement.java b/src/org/lineageos/hardware/ReadingEnhancement.java
index 90839e9..9defe69 100644
--- a/src/org/lineageos/hardware/ReadingEnhancement.java
+++ b/src/org/lineageos/hardware/ReadingEnhancement.java
@@ -39,7 +39,8 @@ public class ReadingEnhancement {
     private static final int LEVEL_COLOR_MATRIX_READING = LEVEL_COLOR_MATRIX_GRAYSCALE + 1;
 
     private static final int MODE_UNSUPPORTED          = 0;
-    private static final int MODE_HWC2_COLOR_TRANSFORM = 1;
+    private static final int MODE_HWC1_COLOR_TRANSFORM = 1;
+    private static final int MODE_HWC2_COLOR_TRANSFORM = 2;
 
     private static final int sMode;
 
@@ -70,7 +71,7 @@ public class ReadingEnhancement {
                     com.android.internal.R.bool.config_setColorTransformAccelerated)) {
             sMode = MODE_HWC2_COLOR_TRANSFORM;
         } else {
-            sMode = MODE_UNSUPPORTED;
+            sMode = MODE_HWC1_COLOR_TRANSFORM;
         }
     }
 
-- 
2.17.0

